### PR TITLE
Get users by `int` or `Guid` ids

### PR DESCRIPTION
**Pull request by bjarnef**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description
With https://github.com/umbraco/Umbraco-CMS/pull/13868 users support have key/guid like content, media etc.

The `UsersController` should allow fetching user/users by `Guid` as well.
Not sure if users will have `Udi` as well in v14?

With the new backoffice I guess the user picker may store udi? (or guid) instead of user id as the current user picker does.

### CODE REVIEW

**Comment by bjarnef** [comment]
@nikolajlauridsen @bergmania some suggestions after https://github.com/umbraco/Umbraco-CMS/pull/13868 has been merged.

@iOvergaard I guess there will be an user picker in the new backoffice as well? and I guess it makes sense to store udi, but it doesn't seem `User` is a valid object/udi type at the moment:
https://github.com/umbraco/Umbraco-CMS/blob/v14/dev/src/Umbraco.Core/Models/UmbracoObjectTypes.cs

Unlike the current member and user pickers, I guess both in the new backoffice also will allow better configuration as single/multi picker, where they currently have some limitations.

File: src/Umbraco.Web.BackOffice/Controllers/UsersController.cs [hunk 1]
```diff
@@ -315,6 +317,25 @@ public ActionResult<string[]> PostClearAvatar(int id)
...
+    /// <summary>
+    ///     Gets a user by Id
+    /// </summary>
+    /// <param name="id"></param>
+    /// <returns></returns>
+    [OutgoingEditorModelEvent]
+    [Authorize(Policy = AuthorizationPolicies.AdminUserEditsRequireAdmin)]
+    public ActionResult<UserDisplay?> GetById(Guid id)
+    {
+        IUser? user = _userService.GetAsync(id).Result;
```
Comment (bjarnef) [inline src/Umbraco.Web.BackOffice/Controllers/UsersController.cs:329]: I guess the method ideally should be `async` instead?

**Comment by emmagarland** [comment]
Hi @bjarnef,

Thanks for your PR relating to #13868, where users can be retrieved by int or Guid.

One of the Core Collaborators team will review this as soon as possible - it also seems interesting in terms of expected behaviour (and you've asked a few questions on this) that we want HQ to review, so I'll contact them about it.

Best wishes

Emma

**Comment by emmagarland** [comment]
Hi @bjarnef 

Just checked in with @bergmania and the old controllers will be removed for v14. The new ones in the management API do support Guids, and should not support Ints.
 He also mentioned that we try to avoid Udi's.

I will therefore close this PR, but appreciate your thoughts on creating it!

Thanks, 

Emma

**Comment by bjarnef** [comment]
Hi @emmagarland 

Okay, makes sense :)
I will check again later in the progress of the new backoffice and Umbraco 14.

I guess the user picker may be a bit different from others pickers storing udi and would probably store guid or comma delimited string of guid in case of a multi user picker. IIRC the current user picker only allow selecting a single user.

**Pull request closed without merging**